File: MML-Card-BE/Models/Card/Creature/Interfaces/IBaseCreature.cs
using MML_Card_BE.Models.Card.Generic.Interfaces;
using MML_Card_BE.Models.Card.Useable;
using MML_Card_BE.Models.Enum;

namespace MML_Card_BE.Models.Card.Creature.Interfaces
{
    public interface IBaseCreature : ICardBase
    {
        int Level { get; set; }
        List<Equipment> EquippedItems { get; set; }
        List<EquipmentSlot> EquipmentSlots { get; set; }
        List<Effect> Effect { get; set; }
        List<Ability> Abilities { get; set; }
        Func<string> Exhaust { get; set; }
    }
}

File: card-web-client/src/models/Creature/IBaseCreature.ts
import { ICardBase } from '../CardBase';
import EquipmentSlot from '../Enum/EquipmentSlotEnum';
import { Effect } from '../Useable';
import Equipment from '../Useable/Equipment';
import Ability from './Ability';

interface IBaseCreature extends ICardBase {
  level: number;
  equippedItems: Equipment[];
  equipmentSlots: EquipmentSlot[];
  effect: Effect[];
  Abilities: Ability[];
  Exhaust: Function;
}

export default IBaseCreature;
